src/db/migrations/migration2.js

```diff
@@ -0,0 +1,120 @@
+const pLimit = require('p-limit');
+const async = require('async');
+const logger = require('../../utils/logger');
+const DBHelper = require('../db-helper');
+const { TX_STATUS, EVENT_STATUS } = require('../../constants');
+const MultipleResultsEventApi = require('../../api/multiple-results-event');
+const EventLeaderboard = require('../../models/event-leaderboard');
+const GlobalLeaderboard = require('../../models/global-leaderboard');
+
+const PROMISE_CONCURRENCY_LIMIT = 15;
+const limit = pLimit(PROMISE_CONCURRENCY_LIMIT);
+
+async function wrapper(dbMethod, callback) {
+  await dbMethod();
+  callback();
+}
+
+async function migration2(next) {
+  try {
+    // migrate event leaderboard
+    const events = await DBHelper.findEvent({ txStatus: TX_STATUS.SUCCESS });
+    const promises = [];
+
+    events.forEach((event, key) => {
+      promises.push(limit(async () => {
+        const eventAddress = event.address;
+        const resultSet = await DBHelper.findResultSet({ eventRound: 0, txStatus: TX_STATUS.SUCCESS, eventAddress: event.address });
+        const bets = await DBHelper.findBet({
+          eventAddress: event.address,
+          txStatus: TX_STATUS.SUCCESS,
+        });
+        const txs = resultSet.concat(bets);
+
+        try {
+          let i = 0;
+          await async.whilst(
+            check => check(null, i < txs.length), // trigger iter
+            (next) => {
```

Comment:
Rename next, this is shadowing the same var as the `async function migration2(next) {`.